### ISSUE TITLE
DAG copy method

### ISSUE THREAD

**Issue by maddy-tod** [Qiskit/qiskit-terra#2046]
### What is the expected enhancement?
The code to create a new DAG and then copy over the metadata from another DAG is used in many passes, e.g. in lookahead swap
```
target_dag = DAGCircuit()
target_dag.name = source_dag.name

for creg in source_dag.cregs.values():
     target_dag.add_creg(creg)
```
It might be good to have this functionality as a method on the original DAG, so you could call `dag.blank_copy()` or similar to retrive a copy of the DAG with the same name/creg/qreg.

**Comment by 1ucian0** [comment]
I think we have to move towards a model where that kind of copy is not needed. The target dag should be extended automatically, with no need to pre-add the regs.

**Comment by 1ucian0** [comment]
closed via PR #4915

https://github.com/Qiskit/qiskit-terra/pull/4915/files#diff-feb434f18758792899804d6e18c57707

**Comment by 1ucian0** [comment]
closed via PR #4915

https://github.com/Qiskit/qiskit-terra/pull/4915/files#diff-feb434f18758792899804d6e18c57707

**Comment by 1ucian0** [comment]
closed via PR #4915

https://github.com/Qiskit/qiskit-terra/pull/4915/files#diff-feb434f18758792899804d6e18c57707
